src/tabpane/TabButton.ts

```diff
@@ -0,0 +1,167 @@
+import { DNode } from '@dojo/widget-core/interfaces';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+
+import * as css from './styles/tabPane.m.css';
+
+/**
+ * @type TabButtonProperties
+ *
+ * Properties that can be set on a TabButton component
+ *
+ * @property active             Determines whether this tab button is active
+ * @property closeable          Determines whether this tab can be closed
+ * @property controls           ID of the DOM element this tab button controls
+ * @property disabled           Determines whether this tab can become active
+ * @property id                 ID of this tab button DOM element
+ * @property index              The position of this tab button
+ * @property onClick            Called when this tab button is clicked
+ * @property onCloseClick       Called when this tab button's close icon is clicked
+ * @property onEndPress         Called when the end button is pressed
+ * @property onHomePress        Called when the home button is pressed
+ * @property onDownArrowPress   Called when the down arrow button is pressed
+ * @property onLeftArrowPress   Called when the left arrow button is pressed
+ * @property onRightArrowPress  Called when the right arrow button is pressed
+ * @property onUpArrowPress     Called when the up arrow button is pressed
+ */
+export interface TabButtonProperties extends ThemeableProperties {
+	active?: boolean;
+	closeable?: boolean;
+	controls: string;
+	disabled?: boolean;
+	id: string;
+	index: number;
+	onClick?: (index: number) => void;
+	onCloseClick?: (index: number) => void;
+	onEndPress?: () => void;
+	onHomePress?: () => void;
+	onDownArrowPress?: () => void;
+	onLeftArrowPress?: () => void;
+	onRightArrowPress?: () => void;
+	onUpArrowPress?: () => void;
+};
+
+export const TabButtonBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TabButton extends TabButtonBase<TabButtonProperties> {
+	private _onClick() {
+		const {
+			disabled,
+			index,
+			onClick
+		} = this.properties;
+
+		!disabled && onClick && onClick(index);
+	}
+
+	private _onCloseClick(event: MouseEvent) {
+		const {
+			index,
+			onCloseClick
+		} = this.properties;
+
+		event.stopPropagation();
+		onCloseClick && onCloseClick(index);
+	}
+
+	private _onKeyDown(event: KeyboardEvent) {
+		const {
+			closeable,
+			disabled,
+			index,
+			onCloseClick,
+			onEndPress,
+			onHomePress,
+			onDownArrowPress,
+			onLeftArrowPress,
+			onRightArrowPress,
+			onUpArrowPress
+		} = this.properties;
+
+		if (disabled) {
+			return;
+		}
+
+		// Accessibility
+		switch (event.keyCode) {
+			// Escape
+			case 27:
```

Comment:
Do you think this would be more readable with a `const ESCAPE_KEY = 27;` instead? If the code needs comments to be understood then it's likely that it could be re-written better.